### DISCUSSION TITLE
`astrojs` LSP configuration

### DISCUSSION THREAD

**Discussion by mangkoran** [joaotavora/eglot#1072] [Q&A]
First of all, thank you for your hard work! I have migrated from `lsp-mode` to `eglot` for the sake of simplicity (and learning `emacs` ofc).

Recently I started to learn [Astro](https://astro.build/). They do have their own L[SP server](https://github.com/withastro/language-tools/blob/main/packages/language-server) for their `*.astro` files. However, I couldn't setup it right. Running `astro-ls --stdio` keep failing.

Do you have any suggestion on what configuration should I do?

**Comment by gpampara** [comment]
Recent versions (not sure from when) of the astro language server rebased the internals of the language server on [volar.js](https://github.com/volarjs/volar.js). This requires `initializationOptions` to be passed to the language server upon startup. I have the following defined to get the language server running:

```
(with-eval-after-load 'eglot
    (add-to-list 'eglot-server-programs
        `(astro-ts-mode . ("${pkgs.nodePackages.astro-language-server}/bin/astro-ls" "--stdio"
                                           :initializationOptions (:typescript (:tsdk "${pkgs.nodePackages.typescript}/lib/node_modules/typescript/lib"))))))
```

Please ignore the nix template variables but you should get the gist.
